src/main/java/com/ncc/aif/ValidateAIF.java

```diff
@@ -306,24 +383,24 @@ public Resource validateKBAndReturnReport(Model dataToBeValidated, Model union)
             engine.setMaxDepth(depth);
             try {
                 engine.applyEntailments();
-                engine.validateAll(executor);
+                reports.addAll(engine.validateAll(executor));
                 validationMetadata = engine.getValidationMetadata();
-                return engine.getReport();
+                lastDuration = engine.getLastDuration();
             } catch (InterruptedException | ExecutionException e) {
                 System.err.println("Unable to validate due to exception");
                 e.printStackTrace();
-                return null;
             }
         } else {
             ValidationEngine engine = ValidationUtil.createValidationEngine(unionModel, shacl, config);
             engine.setProgressMonitor(progressMonitor);
             try {
                 engine.applyEntailments();
-                return engine.validateAll();
+                reports.add(engine.validateAll());
             } catch (InterruptedException ex) {
-                return null;
+                // don't add anything to reports
```

Comment:
It's not clear to me why you changed the validator so that it returns an empty set instead of returning null to signify an engine error, at least in the case of `validateKBAndReturnMultipleReports` (but not `validateKBAndReturnReport`). It seems like null conveys "error" more than an empty set of validation reports does...which if anything seems to convey that everything's valid. At a minimum, you need to update the JavaDoc.

Reply:
That's fair. I've modified this method to return null when an error has occurred. Updated calls to that method accordingly.

Updated JavaDoc.